Pat/BreedList/BreedListInteractor.swift

```diff
@@ -0,0 +1,18 @@
+import Foundation
+
+class BreedListInteractor {
+
+    let dataFetcher = DataFetcher()
+
+    func loadBreedList(completion: @escaping ([Breed]) -> Void) {
+        dataFetcher.fetchData(path: URL.breeds) { (result: Result<[String], Error>) in
+            switch result {
+            case .failure(let error):
+                print("Error trying to fetch data. \(error)")
```

Comment:
A view controller deveria ter acesso ao erro também pra mostrar algum feedback pro usuário, vc quer retornar o erro também agora ou deixar pra quando for fazer esse feedback de erro?

Eu sugeriria receber duas closures diferentes ao invés de uma só, pra facilitar lá na view controller. Tipo:
```swift
func loadBreedList(onSuccess: @escaping ([Breed]) -> Void, onError: @escaping (Error) -> Void) {
...
```